### PR TITLE
add docs for cri-o registry mirror

### LINKED ISSUE #1419: How to use Dragonfly with CRI-O?
## Question

I want to use Dragonfly in my k8s cluster,which used **CRI-O**  container engine:
```
NAME      STATUS   ROLES    AGE   VERSION   INTERNAL-IP     EXTERNAL-IP   OS-IMAGE                KERNEL-VERSION              CONTAINER-RUNTIME
node1      Ready    <none>   21h   v1.16.0      xxx        <none>        CentOS Linux 7 (Core)   5.7.7-1.el7.elrepo.x86_64   cri-o://1.16.6
```
I tried to modify some configurations, but when I pulled the image, it did not use dragonfly.
Where is the problem？

I want to pull images from harbor and docker.io,quay.mirrors.ustc.edu.cn,hub-mirror.c.163.com.etc

dfdaemon.yml
```
apiVersion: apps/v1
kind: DaemonSet
metadata:
  labels: 
    k8s-app: dfdaemon
  name: dfdaemon
  namespace: kube-system
spec:
  selector:
    matchLabels:
      app: dfdaemon
  template:
    metadata:
      labels:
        app: dfdaemon
    spec:
      containers:
      - name: dfdaemon
        image: dragonflyoss/dfclient:1.0.3
        imagePullPolicy: IfNotPresent
        resources:
          requests:
            cpu: 250m
        volumeMounts:
        - mountPath: /etc/dragonfly/dfdaemon.yml
          subPath: dfdaemon.yml
          name: dfconf
        - mountPath: /root/.small-dragonfly
          name: data 
        - mountPath: /certs
          name: certs
        - mountPath: /etc/localtime
          name: localtime
      hostNetwork: true
      dnsPolicy: ClusterFirstWithHostNet
      restartPolicy: Always
      tolerations: 
      - effect: NoExecute
        operator: Exists
      - effect: NoSchedule
        operator: Exists
      volumes:
      - name: dfconf
        configMap:
          name: dfdaemon-conf
          defaultMode: 384
      - name: data
        hostPath: 
          path: /home/eecdn/.small-dragonfly
          type: DirectoryOrCreate
      - name: certs
        hostPath:
           path: /data/dfcert
           type: DirectoryOrCreate 
      - name: localtime
        hostPath: 
          path: /etc/localtime

---
apiVersion: v1
kind: ConfigMap
metadata:
  labels: 
    k8s-app: dfdaemon
  name: dfdaemon-conf
  namespace: kube-system
data:
  dfdaemon.yml: |-
    # proxy all http image layer download requests with dfget
    proxies:
      - regx: blobs/sha256.*
    hijack_https:
      cert: /certs/df.crt
      key: /certs/df.key
      hosts:
        - regx: myharbor.test.com:1443
          insecure: true
    supernodes:                       
      - supernode-ip

```
/etc/containers/registries.conf : I added [registries.insecure] and [registries.mirror] as following
(referring this issue: https://github.com/cri-o/cri-o/issues/372)
```
unqualified-search-registries = ['myharbor.test.com:1443','hub-mirror.c.163.com','docker.io','quay.mirrors.ustc.edu.cn', 'registry.access.redhat.com', 'registry.fedoraproject.org', 'quay.io', 'registry.centos.org']

[[registry]]
prefix = "myharbor.test.com:1443"
insecure = true
blocked = false
location = "localhost:65001"

[[registry]]
prefix = "docker.io/library"
insecure = true
blocked = false
location = "localhost:65001/library"

[[registry.mirror]]
location = "localhost:65001"
insecure = true


```

/usr/lib/systemd/system/crio.service : I added 
```
Environment="HTTP_PROXY=http://127.0.0.1:65001"
Environment="HTTPS_PROXY=http://127.0.0.1:65001"

```

### PR DESCRIPTION
**Pull request by jim3ma**
Signed-off-by: 楚贤 <chuxian.mjj@antfin.com>

### Ⅰ. Describe what this PR did

add docs for cri-o registry mirror configuration.

### Ⅱ. Does this pull request fix one issue?

fix #1419 

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

N/A

### Ⅳ. Describe how to verify it

N/A

### Ⅴ. Special notes for reviews

N/A

### CODE REVIEW

File: docs/user_guide/crio_mirror.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Use Dfdaemon as Registry Mirror for CRI-O
```
Comment (Starnop) [inline docs/user_guide/crio_mirror.md:1]: How about adding a sub dir for using-docs for all container engines?

Comment (jim3ma) [reply docs/user_guide/crio_mirror.md:1]: OK, but we need prepare environment for all container engines testing, it may be cost lost of time.

Comment (Starnop) [reply docs/user_guide/crio_mirror.md:1]: I think we do not need to test for each container engine. If the community has corresponding needs, we are more inclined to welcome members of the community to complete it. However, we should manage it all together.

Comment (jim3ma) [reply docs/user_guide/crio_mirror.md:1]: Good suggestions

**Comment by jim3ma** [comment]
@Starnop PTAL

**Pull request merged**